### PR TITLE
Project labyrinth, Linnéa Helén & Katharina Liebig

### PR DESCRIPTION
**Pull request by katliebig**
Link to my deployed project: https://nes-project-labyrinth.netlify.app/
 
My reflections on how this week's project turned out: We're happy :)

Things I'd like to have clarified or explained in more detail: -

### CODE REVIEW

File: code/src/components/GameScreen.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react'
+import { useSelector } from 'react-redux'
+import "nes.css/css/nes.min.css";
+
+import Action from './Action'
+import ActionButton from './ActionButton'
+import RestartButton from './RestartButton'
+
+const GameScreen = () => {
```
Comment (AsaHildebrand) [inline code/src/components/GameScreen.js:9]: Really nice with this map tile function 🥇

**Review by AsaHildebrand — APPROVED** [review]
A very nice project! We don't have much to add. 😄  Nice complexity and the comments are explaning the code in a good way! Awesome UI and design as well as the interactivity. Really nice that you got the nes.css working and linked! Perhaps could add user validation if you type a super long user name to limit characters. Really good job 👏

File: code/src/components/GameScreen.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  // these are the coordinates for the map, booleans because the class names depend on them
+  // 2 dimensional array of 4x2 (rows x columns)
+  const mapTilesArray = [
+    [false, false],
+    [false, false],
+    [false, false],
+    [false, false]
+  ]
```
Comment (KarolCislowski) [inline code/src/components/GameScreen.js:33]: Great 🦸🏻

File: code/src/components/GameScreen.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  setMapTile(gameState.coordinates)
+
+  return (
+    <div className="nes-container is-rounded is-dark game-screen">
+
+      <div className="action-container">
+        <p><img src={setImagePath()} className="icon" alt="icon" />{gameState.description}</p>
+
+        {gameState.coordinates !== "1,3" && <p>What do you do?</p>}
+        {gameState.coordinates === "1,3" && <RestartButton />}
```
Comment (KarolCislowski) [inline code/src/components/GameScreen.js:73]: It is a nice solution, but it is not a future-proof one. What if the map structure will change on the backend (the game will grow for example)? However, in this case, we can safely assume that we are in the final room if there are **no remaining actions**

So our logic could be something like:

```jsx
if (actions.length === 0) {
   return <GameOver />;
}else{
...
```
or 
```jsx
...
actions.length === 0 && <EndScreen/>
...
```

File: code/src/components/StartScreen.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    <div className="nes-container is-rounded is-dark start-screen">
+      <img src="./assets/cave.png" className="start-screen-icon" alt="icon of a cave" />
+      <h1 className="welcome-text">Welcome Adventurer!</h1>
+      <label htmlFor="dark_field">Enter your name:</label>
+
+      <input
+        type="text"
+        id="dark_field"
+        className="nes-input is-dark"
+        placeholder="Kickass Karol"
```
Comment (KarolCislowski) [inline code/src/components/StartScreen.js:25]: so sweet 😻

File: code/src/components/StartScreen.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react'
+import { useDispatch } from 'react-redux'
+import "nes.css/css/nes.min.css";
```
Comment (KarolCislowski) [inline code/src/components/StartScreen.js:3]: remember to separate css imports with an empty line 😸

**Review by KarolCislowski — APPROVED** [review]
This is an awesome project 😺

**Pull request closed without merging**